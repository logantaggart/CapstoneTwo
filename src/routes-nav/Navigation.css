.nav {
    background-color: seashell;
    border-top: 1.5px ridge black;
    border-bottom: 1.5px ridge black;
}

.imglogo {
    width: 35px;
    border: 1px solid black
}

.lnk {
    color: black
}

.lnk:hover {
    color: green
}