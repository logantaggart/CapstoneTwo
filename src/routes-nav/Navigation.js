import React, { useContext } from 'react'
import { Link, NavLink } from 'react-router-dom'
import UserContext from '../auth/UserContext'

import logo from '../crypticlogo.jpeg'

import './Navigation.css'

function Navigation({ logout }) {
    const { currentUser } = useContext(UserContext)

    function loggedInNav() {
        return (
            <ul className="navbar-nav ml-auto">
                <li className="nav-item mr-4">
                    <NavLink className="nav-link lnk" to="/watchlist">
                        Watchlist
                    </NavLink>
                </li>
                <li className="nav-item mr-4">
                    <NavLink className="nav-link lnk" to="/profile">
                        Profile
                    </NavLink>
                </li>
                <li className="nav-item">
                    <Link className="nav-link lnk" to="/" onClick={logout}>
                        Log Out ({currentUser.firstName || currentUser.username})
                    </Link>
                </li>
            </ul>
        )
    }

    function loggedOutNav() {
        return (
            <ul className="navbar-nav ml-auto">
                <li className="nav-item mr-4">
                    <NavLink className="nav-link lnk" to="/login">
                        Log In
                    </NavLink>
                </li>
                <li className="nav-item mr-4">
                    <NavLink className="nav-link lnk" to="/signup">
                        Sign Up
                    </NavLink>
                </li>
            </ul>
        )
    }

    return (
        <nav className="Navigation navbar navbar-expand-md nav">
            <Link className="navbar-brand nav-link" to="/">
                <img src={logo} className="imglogo" />
            </Link>
            {currentUser ? loggedInNav() : loggedOutNav()}
        </nav>
    )
}

export default Navigation