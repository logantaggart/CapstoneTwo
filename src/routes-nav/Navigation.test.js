import React from 'react'
import { MemoryRouter } from 'react-router'

import { render } from '@testing-library/react'

import UserProvider from '../testUtils'

import Navigation from './Navigation'

it('renders without crashing', () => {
    render((
        <MemoryRouter>
            <UserProvider>
                <Navigation />
            </UserProvider>
        </MemoryRouter>
    ))
})

it('matches snapshot', () => {
    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider>
                <Navigation />
            </UserProvider>
        </MemoryRouter>
    )

    expect(asFragment()).toMatchSnapshot()
})

it('matches snapshot when logged out', () => {
    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider currentUser={null}>
                <Navigation />
            </UserProvider>
        </MemoryRouter>
    )

    expect(asFragment()).toMatchSnapshot()
})