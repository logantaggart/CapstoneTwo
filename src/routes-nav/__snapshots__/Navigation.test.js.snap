// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="Navigation navbar navbar-expand-md nav"
  >
    <a
      class="navbar-brand nav-link"
      href="/"
    >
      <img
        class="imglogo"
        src="crypticlogo.jpeg"
      />
    </a>
    <ul
      class="navbar-nav ml-auto"
    >
      <li
        class="nav-item mr-4"
      >
        <a
          class="nav-link lnk"
          href="/watchlist"
        >
          Watchlist
        </a>
      </li>
      <li
        class="nav-item mr-4"
      >
        <a
          class="nav-link lnk"
          href="/profile"
        >
          Profile
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          class="nav-link lnk"
          href="/"
        >
          Log Out (testuser)
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <nav
    class="Navigation navbar navbar-expand-md nav"
  >
    <a
      class="navbar-brand nav-link"
      href="/"
    >
      <img
        class="imglogo"
        src="crypticlogo.jpeg"
      />
    </a>
    <ul
      class="navbar-nav ml-auto"
    >
      <li
        class="nav-item mr-4"
      >
        <a
          class="nav-link lnk"
          href="/login"
        >
          Log In
        </a>
      </li>
      <li
        class="nav-item mr-4"
      >
        <a
          class="nav-link lnk"
          href="/signup"
        >
          Sign Up
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
